blog/static/img/2022-01-03-pcie_screamer.md

```diff
@@ -0,0 +1,301 @@
+ln -s /usr/local/lib64/libssl.so.1.1  /usr/lib64/libssl.so.1.1
+ln -s /usr/local/lib64/libssl.so.1.1  /usr/lib/libssl.so.1.1
+find / -name libcrypto.so.1.1
+ln -s /home/ubuntu/openssl-1.1.1o/libcrypto.so.1.1    /usr/lib64/libcrypto.so.1.1
+ln -s /home/ubuntu/openssl-1.1.1o/libcrypto.so.1.1     /usr/lib/libcrypto.so.1.1
+```
+After building the openssl library, a command was issued to test the communication
+of the PCIScreamer board with the pcileech application:
+
+```bash
+mgabryelski@maciej-HP:~/PCIScreamer$ sudo ./pcileech probe -device fpga -v
+
+DEVICE: FPGA: ERROR: Unable to load FTD3XX.dll [0,v0.0,0000]
+PCILEECH: Failed to connect to the device.
+mgabryelski@maciej-HP:~/PCIScreamer$ 
+```
+The FTDI for USB 3.0 chip is used to connect the PCIScreamer board with a PC, 
+the driver is not embedded in the Linux kernel. So I compiled the driver
+for the FTDI chip from the sources in this Github repository: [FTDI FT50x driver](https://github.com/ufrisk/pcileech/releases/tag/v4.15) The compilation was successful, 
+and after it was finished we have such files in working directory:
+
+```bash
+mgabryelski@maciej-HP:~/GIT1/pcie_screamer/drivers/ft60x$ ls
+51-ft60x.rules  ft60x.ko   ft60x.mod.c  ft60x.o     ftd3xx.h    Makefile       Module.symvers
+ft60x.c         ft60x.mod  ft60x.mod.o  ftconfig.c  loopback.c  modules.order
+mgabryelski@maciej-HP:
+```
+We try to load kernel module with FTDI driver issuing command:
+
+```bash
+mgabryelski@maciej-HP:~/PCIScreamer$ sudo insmod ft60x.ko
+````
+
+After that the kernel module is loaded:
+
+```bash
+mgabryelski@maciej-HP:~/PCIScreamer$ lsmod
+Module                  Size  Used by
+ft60x                  20480  0
+algif_skcipher         16384  0
+af_alg                 32768  1 algif_skcipher
+binfmt_misc            24576  1
+```
+Now checking `pcileech` application we have;
+```bash
+gabryelski@maciej-HP:~/PCIScreamer$ sudo ./pcileech probe -device fpga -v
+
+DEVICE: FPGA: ERROR: Unable to load FTD3XX.dll [0,v0.0,0000]
+PCILEECH: Failed to connect to the device.
+mgabryelski@maciej-HP:~/PCIScreamer$ 
+```
+As we can see, the problem still exists. I tried also building `pcileech` application 
+from sources following this tutorial [PCILeech on Linux](https://github.com/ufrisk/pcileech/releases/tag/v4.15).
+Unfortunately, all attempts to communicate the `pcileech` application with the 
+PCIScreamer FPGA board ended in failure, therefore, more advanced  tests were 
+not feasible.
+
+## Problems with the stability of the PCIE Screamer board
+
+PCIScreamer in revision R01 (which we have) is known for its stability problems
+and with some motherboards it does not work at all.
+
+The incorrect power delivered to the FPGA could be the result of the faulty PCIe
+low-profile riser card used to connect the PCIeScreamer to the motherboard.
+
+## Perspectives on further work
+
+Further work should include checking the FPGA with a different motherboard,
+preferably without a low-profile riser card. 
```

Comment:
What about problems with `pcileech` command? Failing `pcileech` is probably unrelated to stability problems. Also, are we sure that PCIScreamer is having problem in the board we used? Since, it's detected on PCI I assume there are no stability problems.
Don't we need additional user-space library for FTDI to work? Driver installation guide mentions something about such a library (section 2.1.1 Native Compiling): https://ftdichip.com/wp-content/uploads/2020/08/AN_220_FTDI_Drivers_Installation_Guide_for_Linux-1.pdf

Reply:
@arturkow2000 I think it might be stability issues. In the corrected version of the document, I described an error in communication to the FPGA board by PCI bus (this time the error hasn't been related to FTDI driver error). sorry it was my fault because I didn't post the last trial with board.